vm/src/obj/objstr.rs

```diff
@@ -532,6 +532,22 @@ impl PyString {
         )
     }
 
+    #[pymethod]
+    fn removeprefix(&self, pref: PyStringRef) -> PyResult<String> {
+        if self.value.as_str().starts_with(&pref.value) {
```

Comment:
Basically this is the same as byteinner implementation. If you are interested in, there is a common string implementation file `pystr.rs`. If you don't want to try this change, it is good as it is.

Reply:
Thanks for the hint, I had already realized this also the existance of pystr. All sotions I found here, would lead to very fine granular functions that need to be implemented in specializations of the trait. I have not found a way to a common trait or predicate that offers a starts_with or an equivalent function (at least not without significant overhead). One way would be using Pattern, but that is marked as unstable. Do you know a way to implement that?

Reply:
Because this is simple string match, I think python `str.startswith` is the the similar case. I think adding `py_removeaffix` will be very similar to this one.

- pystr part: https://github.com/RustPython/RustPython/blob/master/vm/src/obj/pystr.rs#L172-L205
- objstr part: https://github.com/RustPython/RustPython/blob/master/vm/src/obj/objstr.rs#L524-L533
- objbytes part: https://github.com/RustPython/RustPython/blob/master/vm/src/obj/objbytes.rs#L287-L300

Please let me know if I missed something you concern about.

Reply:
In that functions, the unified implementation helps fa rmore as the logic is more tricky and there is some more error handling required. I found an acceptable trade off.

Reply:
great! I like this new implementation